Demo/Demo/ViewModels/PayPalViewModel.swift

```diff
@@ -0,0 +1,135 @@
+import UIKit
+import PayPalNativeCheckout
+import PayPalCheckout
+import PaymentsCore
+
+class PayPalViewModel: ObservableObject, PayPalDelegate {
+
+    enum State {
+        case initial
+        case loading(content: String)
+        case mainContent(title: String, content: String, flowComplete: Bool)
+    }
+
+    @Published private(set) var state = State.initial
+
+    private var accessToken = ""
+
+    private var getAccessTokenUseCase = GetAccessToken()
+    private var getOrderIdUseCase = GetOrderIdUseCase()
+    private var getBillingAgreementToken = GetBillingAgreementToken()
+    private var getBATokenWithoutPurchaseUseCase = GetBATokenWithoutPurchase()
+    private var getApprovalSessionTokenUseCase = GetApprovalSessionId()
+    private var getOrderRequestUseCase = GetOrderRequestUseCase()
+    private var payPalClient: PayPalClient?
+
+    func getAccessToken() {
+        Task {
+            state = .loading(content: "Getting access token")
+            guard let token = await getAccessTokenUseCase.execute() else {
+                state = .mainContent(title: "Error", content: "Failed to fetch access token", flowComplete: true)
+
+                return
+            }
+            accessToken = token
+            payPalClient = PayPalClient(config: CoreConfig(accessToken: token, environment: PaymentsCore.Environment.sandbox))
+            payPalClient?.delegate = self
+            state = .mainContent(title: "Access Token", content: accessToken, flowComplete: false)
+        }
+    }
+
+    func retry() {
+        payPalClient = nil
+        accessToken = ""
+        state = .initial
+    }
+
+    func checkoutWithOrder() {
+        startNativeCheckout {
+            let orderRequest = self.getOrderRequestUseCase.execute()
+            await self.payPalClient?.start { createOrderAction in
+                createOrderAction.create(order: orderRequest)
+            }
+        }
+    }
+
+    func checkoutWithOrderId() {
+        startNativeCheckout {
+            let orderId = try await self.getOrderIdUseCase.execute()
+            await self.payPalClient?.start { createOrderAction in
+                createOrderAction.set(orderId: orderId)
+            }
+        }
+    }
```

Comment:
I am confused the difference between these two?

Reply:
`checkoutWithOrderId` is for a server integration. The merchant might have created the order in his server and has the `orderId` to start NXO

`checkoutWithOrder` allows for the merchants to start NXO with an `OrderRequest` object, without a server integration. The creation of the order is handled inside NXO

Reply:
`checkoutWithOrderId` is for a server integration. The merchant might have created the order in his server and has the `orderId` to start NXO

`checkoutWithOrder` allows for the merchants to start NXO with an `OrderRequest` object, without a server integration. The creation of the order is handled inside NXO

Reply:
this needs more discussion, thought process was to provide a convenience function to pass just `orderId` as String instead of passing `createOrderAction` and then setting orderId,
now the question do we need that convinience function or not, if we decide not to have it, then we can remove one of the functions in above